lib/src/shapes/rectangle.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:flutter/widgets.dart';
+
+/// A rectangular shape that takes the size of the parent
+class Rectangle extends StatelessWidget {
+  const Rectangle({
+    super.key,
+    this.fillColor,
+    this.fillGradient,
+    this.strokeColor,
+    this.strokeGradient,
+    this.strokeLineWidth = 1.0,
+  });
+
+  /// A color used for painting the interior of the rectangle.
+  ///
+  /// This is an alternative to [fillGradient].
+  final Color? fillColor;
+
+  /// A gradient used for painting the interior of the rectangle.
+  ///
+  /// This is an alternative to [fillColor].
+  final Gradient? fillGradient;
+
+  /// A color used for painting the outline of the rectangle.
+  ///
+  /// The stroke line is drawn inside the frame of the rectangle, and its
+  /// width is determined by [strokeLineWidth].
+  ///
+  /// This is an alternative to [strokeGradient].
+  final Color? strokeColor;
+
+  /// A gradient used for painting the outline of the rectangle.
+  ///
+  /// The stroke line is drawn inside the frame of the rectangle, and its
+  /// width is determined by [strokeLineWidth].
+  ///
+  /// This is an alternative to [strokeColor].
+  final Gradient? strokeGradient;
+
+  /// The width of the stroke used to paint the outline of the rectangle.
+  ///
+  /// The default is 1.0.
+  final double strokeLineWidth;
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox.expand(
+      child: CustomPaint(
+        painter: _RectanglePainter(
+          fillColor: fillColor,
+          fillGradient: fillGradient,
+          strokeColor: strokeColor,
+          strokeGradient: strokeGradient,
+          strokeLineWidth: strokeLineWidth,
+        ),
+      ),
+    );
+  }
+}
+
+class _RectanglePainter extends CustomPainter {
+  _RectanglePainter({
+    required this.fillColor,
+    required this.fillGradient,
+    required this.strokeColor,
+    required this.strokeGradient,
+    required this.strokeLineWidth,
+  });
+
+  final Color? fillColor;
+  final Gradient? fillGradient;
+  final Color? strokeColor;
+  final Gradient? strokeGradient;
+  final double strokeLineWidth;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    _paintFill(canvas, size, fillColor, fillGradient);
```

Comment:
You probably don't need to pass the painter properties to other methods, because those methods already have access to properties.